lib/matplotlib/cbook.py

```diff
@@ -527,11 +545,7 @@ def process(self, s, *args, **kwargs):
         """
         if s in self.callbacks:
             for cid, proxy in list(six.iteritems(self.callbacks[s])):
-                # Clean out dead references
-                if proxy.inst is not None and proxy.inst() is None:
-                    del self.callbacks[s][cid]
-                else:
-                    proxy(*args, **kwargs)
+                proxy(*args, **kwargs)
```

Comment:
Looking at where the ``ReferenceError``  in ``_BoundMethodProxy.__call__`` gets raised, it only occurs if the method if the weakref has died.  The improved GC auto deletes all traces of the proxy from ``CallbackRegistry`` when this happens. and thus it should never raise such an error here, unless you think a thread could spark a race-condition (I have very limited knowledge of threads and thread safety).